maintenance/monitor-component-metrics.md

```diff
@@ -3,43 +3,261 @@ title: Monitor Calico component metrics
+```bash
+kubectl apply -f - <<EOF
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRoleBinding
+metadata:
+  name: prometheus-rbac
+subjects:
+  - kind: ServiceAccount
+    name: default
+    namespace: default
+roleRef:
+  kind: ClusterRole
+  name: cluster-admin
+  apiGroup: rbac.authorization.k8s.io
+EOF
+```
+## 2. Creating permanent config file
 
-You can configure Felix and/or Typha to provide metrics to Prometheus.
+Since containers are ephemeral, it is best to store configuration file to a permanent storage solution and link it to your pod. 
 
+> **Note**: A comprehensive guide about configuration file can be [found at this link](https://prometheus.io/docs/prometheus/latest/configuration/configuration/).
+   {: .alert .alert-info}
 
-### How to
+```bash
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: prometheus-config
+data:
+  prometheus.yml: |-
+    global:
+      scrape_interval:   15s
+      external_labels:
+        monitor: 'tutorial-monitor'
+    scrape_configs:
+    - job_name: 'prometheus'
+      scrape_interval: 5s
+      static_configs:
+      - targets: ['localhost:9090']
+    - job_name: 'felix_metrics'
+      scrape_interval: 5s
+      scheme: http
+      kubernetes_sd_configs:
+      - role: endpoints
+      relabel_configs:
+      - source_labels: [__meta_kubernetes_service_name]
+        regex: felix-demo-svc
+        replacement: $1
+        action: keep
+    - job_name: 'typha_metrics'
+      scrape_interval: 5s
+      scheme: http
+      kubernetes_sd_configs:
+      - role: endpoints
+      relabel_configs:
+      - source_labels: [__meta_kubernetes_service_name]
+        regex: typha-demo-svc
+        replacement: $1
+        action: keep
+EOF
+```
 
-#### Enable Prometheus metrics for and Felix and Typha
+## 3. Creating a service to expose Felix metrics
 
-1. Using the Prometheus documentation, configure one or more [Prometheus servers](https://prometheus.io/docs/introduction/overview/).  
-1. To enable [Felix]({{ site.baseurl }}/reference/felix/configuration) for metrics, set **PrometheusMetricsEnabled = true**.
-1. To enable [Typha]({{ site.baseurl }}/reference/typha/configuration) for metrics, set **PrometheusMetricsEnabled = true**.
-1. If required for Felix and/or Typha, change the default TCP port (9091) for your Prometheus metrics server using the parameter, **PrometheusMetricsPort**.
+Your prometheus configuration is going to check services and catch endpoints related to felix-demo-svc service, you need to create a service with that name to expose Felix metrics. This will help in discovery of your services and **makes** it dynamic. 
 
-#### Best practices
+> **Note**: Felix uses port 9091 TCP to publish its metrics.
+   {: .alert .alert-info}
+
+``` bash
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Service
+metadata:
+  name: felix-demo-svc
```

Comment:
Let's use names that someone could run with in production.  So suggest changing "felix-demo-svc" -> "felix-metrics-svc".  And same for typha etc.

Reply:
Great suggestion!
implemented.